### PR TITLE
Clean code guide : Markdown Version

### PR DESCRIPTION
**Pull request by priyasingh199885**
1. Mermaid diagrams
2. .adoc  --> .md
Changes to package.json -- still working (can be discussed in upcoming call)

This PR is only to pre-view the changes and not necessarily be approved or merged.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,11 +19,6 @@
   },
   "homepage": "https://github.com/SAP/styleguides#readme",
   "dependencies": {
-    "@asciidoctor/cli": "^3.5.0",
-    "@asciidoctor/core": "^2.2.6",
-    "asciidoc-link-check": "^1.0.15",
-    "asciidoctor-kroki": "^0.15.4",
-    "asciidoctor-pdf": "1.0.0-alpha.16",
-    "asciidoctor-plantuml": "^1.5.0"
+    "mermaid":"^8.11.0",
```
Comment (xtough) [inline package.json]: Why do we need a dependency on mermaid? We do not want to process mermaid content during build.
We only need a markdown-linkcheck here.

File: abap-code-review/ABAPCodeReview.md [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+The example is provided as a starting point, different organizations have different requirements and work processes, the example can be adjusted to fit any requirements.
+
+## 6.2. Two Way
+
+The developer uses the normal UI of abapGit, [pushing and pulling](https://docs.abapgit.org/) all changes.
+
+If git is the only destination for the code, suggest disabling CTS transports to make it faster to do changes, eg. by developing in local packages.
+<div style="align:center"> 
+   <p style="text-align:center;">
+      <img src="../abap-code-review/images/abapgit1_107_0.png"  align="middle" width="85%" height="75%" /></p>
```
Comment (xtough) [inline abap-code-review/ABAPCodeReview.md:366]: same here

**Review by xtough — CHANGES_REQUESTED** [review]
Nice progress! Some points to check:
* [ ] all abap-code-review/src/*.md files should be deleted because everything is contained in one single abap-code-review/ABAPCodeReview.md file
* [ ] markdown-linkcheck should be integrated in package.json scripts
* [ ] README.md and abap-code-review/README.md need to be adjusted to new files and build process

File: package.json (hunk 1 above)
Comment (priyasingh199885) [reply package.json]: Yes, mermaid is removed. Do we also delete the [asciidoc-link-check-config.json](https://github.com/priyasingh199885/styleguides/blob/main/asciidoc-link-check-config.json) file?

**Comment by priyasingh199885** [comment]
> Nice progress! Some points to check:
> 
> * [ ]  all abap-code-review/src/*.md files should be deleted because everything is contained in one single abap-code-review/ABAPCodeReview.md file
> * [ ]  markdown-linkcheck should be integrated in package.json scripts
> * [ ]  README.md and abap-code-review/README.md need to be adjusted to new files and build process

**Comment by priyasingh199885** [comment]
What are the changes needed in README.md? and abap-code-review/Readme.md too?

**Comment by xtough** [comment]
> What are the changes needed in README.md? and abap-code-review/Readme.md too?

* Links to *.adoc and *.pdf need to be changed to the new ABAPCodeReview.md
* build process description for producing PDF needs to be removed / adapted to describe only markdown-linkcheck (once this is implemented)

**Comment by xtough** [comment]
build is currently failing:

> `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.

File: package.json [hunk 3]
```diff
@@ -5,8 +5,8 @@
   "description": "Collect ideas and approaches for reviewing ABAP code.",
   "main": "index.js",
   "scripts": {
-    "links": "find abap-code-review/src -name \\*.adoc | xargs -n1 asciidoc-link-check -c asciidoc-link-check-config.json -p",
-    "test": "npm run links && asciidoctor-web-pdf -r asciidoctor-plantuml -r asciidoctor-kroki -d book -D build abap-code-review/src/index.adoc -a stylesheet=\"asciidoctor-pdf/css/asciidoctor.css,asciidoctor-pdf/css/document.css,asciidoctor-pdf/css/features/title-page.css,custom.css\""
+    "links": "find . -path ./node_modules -prune -false -o -name '*.md' -exec markdown-link-check -c mlc_config.json {} \\;",
```
Comment (xtough) [inline package.json:8]:
```suggestion
    "test": "find . -path ./node_modules -prune -false -o -name '*.md' -exec markdown-link-check -c mlc_config.json {} \\;",
```
no need to redirect `test` to `npm run links` anymore --> rename script `links` to `test`

**Review by xtough — CHANGES_REQUESTED** [review]
Almost there :-)
See suggestion on package.json

**Review by xtough — CHANGES_REQUESTED** [review]
Well done!
Now we just need to replace the links to *.pdf and *.adoc in README.md and abap-code-review/README.md with some reworked text pointing directly at abap-code-review/ABAPCodeReview.md

**Review by xtough — APPROVED** [review]
Looks good to me. 
We can merge this now.

**Comment by KlausHaeuptleSAP** [comment]
@priyasingh199885: Thanks! Great contribution!

**Pull request merged**